### PR TITLE
Perform the true Chef run in Travis CI (linux and mac environments)

### PR DESCRIPTION
**Pull request by rjaros87**
Major changes:
- Added Mac OS X support #25 
- Added test for chef-client in zero mode on Travis #28 

Minor changes:
- Increase Ruby version for workstation to 2.2.2
- Corrected the Berksfile
- Added bats test (it could be also used for the kitchen test)

### CODE REVIEW

**Comment by rjaros87** [comment]
@gildegoma please review. First three commits are from in previous pull request #29 whicha are necessary in this PR

**Comment by gildegoma** [comment]
@rjaros87 this looks great again! It actually resolves some points I also hit while I started to review your #29. 
I'd really love to get this in very soon, but having all the things together might unfortunately slow down the review/test process... I try to come back to you as soon as possible, maybe with suggestion to split/postpone a little bit. But don't worry, we'll get there for sure!

**Comment by gildegoma** [comment]
@rjaros87 don't move for the moment, I'll give you a wish of roadmap ;-)

**Comment by rjaros87** [comment]
Ok. If you merge #29 , from this PR should disappear first three commits automatically and should stay visible only new one.

**Comment by rjaros87** [comment]
> I'll give you a wish of roadmap ;-)

Windows support? ;D

**Comment by rjaros87** [comment]
Little update  - README.md

File: recipes/default.rb [hunk 1]
```diff
@@ -151,16 +176,16 @@
...
+    mode '0755'
   end
 end
 %w(android-update-sdk).each do |android_helper_script|
   template File.join(node['android-sdk']['scripts']['path'], android_helper_script) do
     source "#{android_helper_script}.erb"
     owner node['android-sdk']['scripts']['owner']
-    group node['android-sdk']['scripts']['group']
-    mode 0755
+    group node['android-sdk']['scripts']['group'] unless mac_os_x?
```
Comment (Brantone) [inline recipes/default.rb]: Why the limit on `mac_os_x` ?

**Comment by Brantone** [comment]
Curious what's holding up the merge?

File: recipes/default.rb (hunk 1 above)
Comment (rjaros87) [inline recipes/default.rb]: I had some trouble to set properly group on mac, but now it doesn't occurs. I will remove `unless mac_os_x?` today and recommit to branch.

**Comment by gildegoma** [comment]
Closed in favour of #33 (including these changes as well).

**Pull request closed without merging**